modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/resolver/OrtResultArtifactResolver.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.ort.resolver;
+
+import com.here.ort.model.*;
+import com.here.ort.model.Package;
+import com.here.ort.model.config.PathExclude;
+
+import org.eclipse.sw360.antenna.model.artifact.Artifact;
+import org.eclipse.sw360.antenna.model.artifact.facts.*;
+import org.eclipse.sw360.antenna.model.artifact.facts.dotnet.DotNetCoordinates;
+import org.eclipse.sw360.antenna.model.artifact.facts.java.MavenCoordinates;
+import org.eclipse.sw360.antenna.model.artifact.facts.javaScript.JavaScriptCoordinates;
+import org.eclipse.sw360.antenna.model.xml.generated.MatchState;
+import org.eclipse.sw360.antenna.util.LicenseSupport;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class OrtResultArtifactResolver implements Function<Package, Artifact> {
+    private SortedMap<Identifier, Map<LicenseFinding, List<PathExclude>>> licenseFindings;
+
+    public OrtResultArtifactResolver(OrtResult result) {
+        licenseFindings = result.collectLicenseFindings(false);
+    }
+
+    @Override
+    public Artifact apply(Package pkg) {
+        Artifact a = new Artifact("OrtResult")
+                .addFact(new ArtifactMatchingMetadata(MatchState.EXACT));
+
+        mapCoordinates(pkg).ifPresent(a::addFact);
+        mapSourceUrl(pkg).ifPresent(a::addFact);
+
+        mapDeclaredLicense(pkg).ifPresent(a::addFact);
+        mapFilename(pkg).ifPresent(a::addFact);
+        mapHomepage(pkg).ifPresent(a::addFact);
+
+        mapObservedLicense(pkg).ifPresent(a::addFact);
+        mapCopyrights(pkg).ifPresent(a::addFact);
+
+        return a;
+    }
+
+    private static Optional<ArtifactCoordinates> mapCoordinates(Package pkg) {
+        String namespace = pkg.getId().getNamespace();
+        String name = pkg.getId().getName();
+        String version = pkg.getId().getVersion();
+
+        switch (pkg.getId().getType().toLowerCase()) {
+            case "nuget":
+            case "dotnet":
+                return Optional.of(mapDotNetCoordinates(name, version));
+            case "maven":
+                return Optional.of(mapMavenCoordinates(namespace, name, version));
+            case "npm":
+                return Optional.of(mapJavaScriptCoordinates(name, version));
+            default:
+                return Optional.of(mapSimpleCoordinates(name, version));
```

Comment:
if `namespace != null` one could maybe use `return Optional.of(mapSimpleCoordinates(namespace + ":" + name, version));`. But that is potentially not a good heuristic.

Reply:
This code was just moved from how / where it was before my commit, so I'd like to keep this as-is for now.